pandas/core/generic.py

```diff
@@ -3770,10 +3773,12 @@ def as_matrix(self, columns=None):
         --------
         pandas.DataFrame.values
         """
+        warnings.warn("method ``as_array`` will be removed in a future version. "
+                      "Use ``values`` instead.", FutureWarning, stacklevel=2)
         self._consolidate_inplace()
         if self._AXIS_REVERSED:
-            return self._data.as_matrix(columns).T
```

Comment:
this should just return ``self.values``

Reply:
Returning self.values implies that ``columns=None`` which isn't necessary true for user code.